aiokafka/group_coordinator.py

```diff
@@ -0,0 +1,847 @@
+import asyncio
+import collections
+import logging
+
+import kafka.common as Errors
+from kafka.coordinator.assignors.roundrobin import RoundRobinPartitionAssignor
+from kafka.coordinator.protocol import ConsumerProtocol
+from kafka.common import OffsetAndMetadata, TopicPartition
+from kafka.protocol.commit import (
+    GroupCoordinatorRequest,
+    OffsetCommitRequest_v2 as OffsetCommitRequest,
+    OffsetFetchRequest_v1 as OffsetFetchRequest)
+from kafka.protocol.group import (
+    HeartbeatRequest, JoinGroupRequest,
+    LeaveGroupRequest, SyncGroupRequest)
+
+from aiokafka import ensure_future
+
+log = logging.getLogger(__name__)
+
+
+class AIOConsumerCoordinator(object):
+    """
+    AIOConsumerCoordinator implements group management for single group member
+    by interacting with a designated Kafka broker (the coordinator). Group
+    semantics are provided by extending this class.  See ConsumerCoordinator
+    for example usage.
+
+    From a high level, Kafka's group management protocol consists of the
+    following sequence of actions:
+
+    1. Group Registration: Group members register with the coordinator
+       providing their own metadata
+       (such as the set of topics they are interested in).
+
+    2. Group/Leader Selection: The coordinator select the members of the group
+       and chooses one member as the leader.
+
+    3. State Assignment: The leader collects the metadata from all the members
+       of the group and assigns state.
+
+    4. Group Stabilization: Each member receives the state assigned by the
+       leader and begins processing.
+    """
```

Comment:
We could mention something like: between each phase coordinator awaits all clients to respond. If some do not respond in time - it will revoke their membership